src/EFCore/Infrastructure/EntityFrameworkServicesBuilder.cs

```
@@ -265,6 +266,8 @@ public virtual EntityFrameworkServicesBuilder TryAddCoreServices()
             TryAdd<IQueryTranslationPreprocessorFactory, QueryTranslationPreprocessorFactory>();
             TryAdd<IQueryTranslationPostprocessorFactory, QueryTranslationPostprocessorFactory>();
 
+            TryAdd(p => p.GetService<IDbContextOptions>()?.FindExtension<CoreOptionsExtension>().SimpleLogger ?? new NullSimpleLogger());
```

FindExtension<CoreOptionsExtension>() can return null. This is what happens in Cosmos tests

---

@AndriySvyryd Any idea why this doesn't fail running the test locally? I have the Cosmos emulator running.

---

Nop, fails for me when running from VS

---

@AndriySvyryd What's the best way to debug why Cosmos tests are not running even though the emulator is running?

---

@AndriySvyryd Figured out what is happening. The test runner is failing in  `CosmosDbConfiguredConditionAttribute`. With the command line runner this is just silently ignored and the tests don't run.

---

You can put a breakpoint at https://github.com/aspnet/EntityFrameworkCore/blob/0d76bbf45a42148924b413ef8f37bf49c1ce10d3/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs#L36
But this should fails even without the emulator (that's why the PR check is failing)

---

The real issue here is that if this fails, then there is no indication at all from the command-line runner that anything went wrong, which is pretty sucky.

---

We could file an issue to make the command-line runner to fail on non-null return codes on Windows too, but I faintly remember that there were some issues with this.

---

Filed #19064